src/core/mac/mac.cpp

```diff
@@ -761,44 +784,64 @@ ThreadError Mac::ProcessReceiveSecurity(Frame &aFrame, const Address &aSrcAddr,
         ExitNow();
     }
 
-    VerifyOrExit(aNeighbor != NULL, error = kThreadError_Security);
-
     aFrame.GetSecurityLevel(securityLevel);
     aFrame.GetFrameCounter(frameCounter);
 
-    GenerateNonce(aSrcAddr.mExtAddress, frameCounter, securityLevel, nonce);
+    aFrame.GetKeyIdMode(keyIdMode);
 
-    tagLength = aFrame.GetFooterLength() - Frame::kFcsSize;
+    switch (keyIdMode)
+    {
+#if OPENTHREAD_ENABLE_JOINER
```

Comment:
We do not need this macro here, even wants to enable/disable commissioning when needed.